types/index.d.ts

```diff
@@ -49,6 +49,8 @@ export interface FastifySwaggerUiOptions {
    * route hooks
    */
   uiHooks?: FastifySwaggerUiHooksOptions
+
+  transformSwagger?: (req: FastifyRequest, reply: FastifyReply, swaggerObject: Record<string, any>) => any
```

Comment:
Should I maybe remove the reply?
Should I put the swaggerObject as first parameter? There could be cases where the transformation does not need to be based on the request or reply. So we could put it as the first parameter and if we later need e.g. the fastify instance, we can add it as fourth parameter without breaking implementations.

Reply:
I only need access to the hostname to solve my original problem of multiple tenants.

The swaggerOjbect is always needed, but the rest of the params maybe not, so I think perhaps it would make sense to put swaggerOjbect  first.